tests/code-editor.spec.ts

```diff
@@ -0,0 +1,248 @@
+import { test, expect } from '@playwright/test';
+
+test('Renders JSON code editor correctly', async ({ page }) => {
+  await page.goto('/code-editor-test.html');
+  const validJSONEditor = page.getByTestId('code-editor-json-valid')
+  await expect(validJSONEditor).toHaveCount(1)
+  await expect(validJSONEditor).toHaveAttribute('language', 'json')
+  await expect(validJSONEditor).toHaveAttribute('value', '{"component": "JSON"}')
+  const monacoContainer = validJSONEditor.locator('div').first()
+  await expect(monacoContainer).toHaveCount(1);
+  await expect(monacoContainer).toHaveAttribute('data-mode-id', 'json')
+
+  // TODO: Add language syntax -- should we be checking that the keys and the values are in the 
+});
+
+// Value: JSON invalid
+
+test('Renders JSON code editor that is invalid', async ({page}) => {
+    await page.goto('/code-editor-test.html');
+    const validJSONEditor = page.getByTestId('code-editor-json-invalid')
+    await expect(validJSONEditor).toHaveCount(1)
+    await expect(validJSONEditor).toHaveAttribute('language', 'json')
+    await expect(validJSONEditor).toHaveAttribute('value', '{"component"}')
+    const monacoContainer = validJSONEditor.locator('div').first()
+    await expect(monacoContainer).toHaveCount(1);
+    await expect(monacoContainer).toHaveAttribute('data-mode-id', 'json')
```

Comment:
I would add a check to ensure that the invalid portion of the "value" attribute value should be highlighted as an error. Not sure if there is an attribute you can check for this but I recommend loading the code-editor-test.vue file in a browser and inspecting the invalid code editor element to seesee

Reply:
I would add this to the other invalid cases as well

Reply:
yeah, so i added a list of some of the issues with testing monaco on the general comment but checking for syntactical errors is quite difficult to do from a testing standpoint as monaco doesn't add human readable css classnames or attributes to pieces of text that they have highlighted as an error. 

I spent some time looking at this and couldn't figure out how to make a meaningful test for this, so decided that the only real thing that we could test was that the text was actually rendered even if its not syntactically correct.

Reply:
One thing you could add is a check for the text color styling. It looks like the class mtk8 is added to any text that is styled (See screenshot). This could allow you to check for this class in cases where the text is valid and ensure the class is not applied when the text is not valid. 
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/56745262/224744496-167a1a47-012c-41a6-a730-ff417a8a6e70.png">

Reply:
I added a check for valid language syntax highlighting from what i could see for the highlighting on the code-editor-test page. I left out most of the invalid highlighting because most of them are done via underlines and its unclear what class is actually setting that.